napari/utils/progress.py

```diff
@@ -57,6 +55,7 @@ class progress(tqdm):
     """
 
     monitor_interval = 0  # set to 0 to disable the thread
+    all_progress = EventedSet()  # track all currently active progress objects
```

Comment:
I maybe don't quite get this. `progress` itself in some sense corresponds to one progress bar right? why is it that then that one progress bar should keep track of all the other progress bars?

i.e. if i make three progress bars, will each of them have an `all_progress` attribute that is the same (or different?) evented set that contains all three?

maybe this should just be a new object that isn't an attribute here. i guess the problem with that approach is that there might be no way to connect to register to it in the first place, unless maybe that object is like a global registry of progress bars that is accessed in the progress init

Reply:
> why is it that then that one progress bar should keep track of all the other progress bars?

It's a class attribute, not an instance attribute.  Note how "self" is added to progress.all_progress (not self.all_progress).  I think a class attribute is preferable to an independent object to track instances of progress (and this is a common pattern, see how dask tracks Callback instance for example).

Reply:
Adding docs around this seems important.  I think this is a nice design, but it needs to be called out.  The fact that the progress() constructor has a side-effect - it's added to a singleton, evented collection - needs to be pointed out in the constructor docs.

Reply:
ahhhh ok - totally missed that, but makes total sense. Class attribute sounds like a good way to go. I like it.

Unrelated to this PR, but do we want `viewer.all_viewers` where `all_viewers` is a class attribute and an evented set. I know in the past i've resisted that idea, but maybe that's a nice pattern to use there too. Can discuss elsewhere, but wanted to flag now

Reply:
lol... yes this is exactly where that came up before.  we wanted to make `current()` a classmethod on `Viewer`, but you found it odd that you could do `viewer.current()` on an instance and not always get back the same instance on which it was called.  So we tucked it away on _QMainWindow, ... and thats how napari.current_viewer works.

I'd be supportive of adding it to Viewer... would be much cleaner.

Reply:
> we wanted to make current() a classmethod on Viewer, but you found it odd that you could do viewer.current() on an instance and not always get back the same instance on which it was called.

Indeed, indeed. Potentially irrationally, I still don't like `viewer.current()`. However, `viewer.all_viewers.current()` makes much more sense to me. I'm certainly not trying to be pedantic here, but publicly exposing the set of all viewers on a viewer, and then allowing users to find the current viewer through that set of all viewers makes a lot more sense to me then straight up `viewer.current`.

What do you think? Is @jni on board with `viewer.all_viewers` and `viewer.all_viewers.current()`?